### PR TITLE
Add "today" filter for list

### LINKED ISSUE #87: Time is never parsed for AddCommand
### Description
- When time is provided in the `DD-MM-YYY HHmm` format, the time is not parsed and the "time" component of the task defaults to 00:00

### Action to be taken
- Parsing should be investigated

### PR DESCRIPTION
**Pull request by remo5000**
### Features
- `list` command takes in a `b/` argument to list all tasks due before today (semantics: `b/today` -> "due before today")
- Fixes #87

### CODE REVIEW

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -77,7 +77,23 @@ Format:
...
+
+ * `month`
+
++Example:
+`list`
+
++Example:
+`list b/today`
+
+-Example:
```
Comment (pengnam) [inline docs/UserGuide.adoc]: So is there an all-time option? Its not in the list above.

File: src/main/java/seedu/address/model/task/DueDateIsBeforeTodayPredicate.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    public boolean test(Task person) {
+        /** Get today's date, 2359 */
+        Calendar date = new GregorianCalendar();
+        date.set(Calendar.HOUR_OF_DAY, 23);
+        date.set(Calendar.MINUTE, 59);
+        date.set(Calendar.SECOND, 0);
+        date.set(Calendar.MILLISECOND, 0);
+
+        return person.getDueDate().valueDate.before(date.getTime());
+
```
Comment (pengnam) [inline src/main/java/seedu/address/model/task/DueDateIsBeforeTodayPredicate.java]: Excess new line

File: src/main/java/seedu/address/model/task/DueDateIsBeforeTodayPredicate.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof DueDateIsBeforeTodayPredicate); // instanceof handles nulls
+
```
Comment (pengnam) [inline src/main/java/seedu/address/model/task/DueDateIsBeforeTodayPredicate.java]: Excess new line

**Comment by pengnam** [comment]
Not doing list before week/month yet?

It might also be better to list tasks due before today **and  not completed.**

File: src/main/java/seedu/address/model/task/DueDateIsBeforeTodayPredicate.java (hunk 2 above)
Comment (remo5000) [reply src/main/java/seedu/address/model/task/DueDateIsBeforeTodayPredicate.java]: Good catch 👍

**Pull request merged**